summingbird-storm-test/src/test/scala/com/twitter/summingbird/storm/TopologyTests.scala

```diff
@@ -101,6 +101,87 @@ class TopologyTests extends WordSpec {
     assert(stormTopo.get_spouts_size() == 1)
   }
 
+  /*
+  *  Test Case to check the opt-in mechanism on the FlatMapperProducer.
+  *  Asserts : the parallelism of the source node
+  *            asserts it is a one spout and one bolt topology
+  *            assert the summer parallelism too.
+  */
+  "Opt-in for the flatmap to merge in source." in {
+    val fmNodeName = "flatMapper"
+    val smNodeName = "summer"
+    val sourceNodeName = "source"
+    val p = Storm.source(TraversableSpout(sample[List[Int]])).name(sourceNodeName)
+      .flatMap(testFn).name(fmNodeName)
```

Comment:
can we test a fanout case? I'm worried about that:
```scala
val p1 = Storm.source(spout).flatMap(fn)
val p2 = p1.sumByKey(store)
val p3 = p1.map(g).sumByKey(store2)
p2.also(p3)
```
I'm afraid we aren't checking that the summers are not the only dependants, and hence the others won't be expecting the summer format.